lib/vmpooler/vsphere_helper.rb

```diff
@@ -12,6 +12,60 @@ def initialize(_vInfo = {})
                                          insecure: true
     end
 
+    def add_disk(vmname, size, datastore)
+      begin
+        @connection.serviceInstance.CurrentTime
+      rescue
+        initialize
+      end
+
+      vm = find_vm(vmname) || find_vm_heavy(vmname)[vmname]
```

Comment:
This is called by `pool_manager` and the `host` or `vm` object already exists from the calling method. Can `vmname` be changed to a VM object instead?